packages/edit-site/src/components/sidebar-navigation-screen-page/page-details.js

```diff
@@ -135,7 +140,7 @@ export default function PageDetails( { id } ) {
 				templateTitle: _templateTitle,
 			};
 		},
-		[ record ]
+		[ record, templates ]
```

Comment:
The hook only needs `record?.parent` as a dependency. There's no need to pass whole object.